woodwork/schema.py

```diff
@@ -181,6 +182,91 @@ def _set_index_tags(self, index):
+        """Return list of columns filtered in specified way. In case of collision, favors logical types
+        then semantic tag then column name.
+
+        Args:
+            include (str or LogicalType or list[str or LogicalType]): parameter or list of parameters to
+                filter columns by. Can be Logical Types or Semantic Tags.
+
+        Returns:
+            List[str] of column names that fit into filter.
+        """
+        if not isinstance(include, list):
+            include = [include]
+
+        ltypes_used = set()
+        ltypes_in_schema = {_get_ltype_class(col['logical_type']) for col in self.columns.values()}
+
+        tags_used = set()
+        tags_in_schema = {tag for col in self.columns.values() for tag in col['semantic_tags']}
+
+        cols_to_include = set()
+
+        for selector in include:
+            if _get_ltype_class(selector) in ww.type_system.registered_types:
+                if selector not in ww.type_system.registered_types:
+                    raise TypeError(f"Invalid selector used in include: {selector} cannot be instantiated")
+                if selector in ltypes_in_schema:
+                    ltypes_used.add(selector)
+            elif isinstance(selector, str):
+                # If the str is a viable ltype, it'll take precedence
+                # but if it's not present, we'll check if it's a tag
+                ltype = ww.type_system.str_to_logical_type(selector, raise_error=False)
+                if ltype and ltype in ltypes_in_schema:
+                    ltypes_used.add(ltype)
+                    continue
+                elif selector in tags_in_schema:
+                    tags_used.add(selector)
+            else:
+                raise TypeError(f"Invalid selector used in include: {selector} must be either a string or LogicalType")
+
+        for col_name, col in self.columns.items():
+            if _get_ltype_class(col['logical_type']) in ltypes_used or col['semantic_tags'].intersection(tags_used):
+                cols_to_include.add(col_name)
+
+        return list(cols_to_include)
+
+    def _get_subset_schema(self, subset_cols):
+        '''
+        Creates a new Schema with specified columns, retainig typing information.
+
+        Args:
+            subset_cols (list[str]): subset of columns from which to create the new Schema
+        Returns:
+            Schema: New Schema with attributes from original Schema
+        '''
+        new_logical_types = {}
+        new_semantic_tags = {}
+        new_column_descriptions = {}
+        new_column_metadata = {}
+        for col_name, col in self.columns.items():
+            if col_name not in subset_cols:
+                continue
+            new_logical_types[col_name] = col['logical_type']
+            new_semantic_tags[col_name] = col['semantic_tags']
+            new_column_descriptions[col_name] = col['description']
+            new_column_metadata[col_name] = col['metadata']
```

Comment:
Can we flip this loop around to loop over subset_cols instead of self.columns.items()? If we had a very wide schema - say 100 columns and we only wanted two of them, we would loop over 98 columns we don't care about. Will this work instead so we only loop over the columns we know we care about?
```python
        for col_name in subset_cols:
            col = self.columns[col_name]
            new_logical_types[col_name] = col['logical_type']
            new_semantic_tags[col_name] = col['semantic_tags']
            new_column_descriptions[col_name] = col['description']
            new_column_metadata[col_name] = col['metadata']
```

Reply:
that makes a lot of sense! Changing